src/components/terminal/component_terminal.gd

```diff
@@ -131,7 +141,24 @@ func _on_cli_execute_streamed_data(data:String) -> void:
 	if data.left(3) == "401":
 		_g.popup_manager.open_popup("ConnectPopup")
 		return
-	console.append_bbcode(str(data))
+		
+	var tick = OS.get_ticks_msec()
+	
+	data_chunks[chunk_idx] = data
+	chunk_idx += 1
+	line_count += data.count("\n")
+	
+	if line_count > write_line_count:
+		data_chunks.resize(chunk_idx)
+		console.append_bbcode(data_chunks)
+		data_chunks.resize(max_data_chunks)
+		chunk_idx = 0
+		line_count = 0
+		
+	elif chunk_idx == max_data_chunks:
+		console.append_bbcode(data_chunks.join(""))
```

Comment:
Maybe I'm not fully understanding the logic here... but shouldn't `data_chunks` be cleared here? Or are you skipping it to avoid memory re-allocation?